File: poh5/poh5_4b.scala
//usr/bin/env scala $0 $@;exit
object Main{
	def main(args:Array[String]){
		val Array(w,h,q)=readLine().split(" ").map(_.toInt)
		val m=Array.ofDim[Int](h,w)
		val imos=Array.ofDim[Int](h+1,w+1)
		for(i<-0 until h){
			m(i)=readLine().split(" ").map(_.toInt)
		}
		var s=0
		for(i<-0 until q){
			val Array(w1,h1,w2,h2)=readLine().split(" ").map(_.toInt)
			imos(h1-1)(w1-1)+=1
			imos(h1-1)(w2)-=1
			imos(h2)(w1-1)-=1
			imos(h2)(w2)+=1
		}
		for(i<-0 until h)for(j<-1 until w)imos(i)(j)+=imos(i)(j-1)
		for(i<-1 until h)for(j<-0 until w)imos(i)(j)+=imos(i-1)(j)
		for(i<-0 until h)for(j<-0 until w)if(imos(i)(j)>0)s+=m(i)(j)
		println(s)
	}
}

File: poh5/poh5_4b.swift
//usr/bin/env swift $0 $@;exit
var _a=readLine()!.characters.split{$0==" "}
let W=Int(String(_a[0]))!,H=Int(String(_a[1]))!,Q=Int(String(_a[2]))!
var m=[[Int]](repeating:[Int](repeating:0,count:W),count:H)
var imos=[[Int]](repeating:[Int](repeating:0,count:W+1),count:H+1)
var h=0,w=0,i=0
for h in 0..<H {
	_a=readLine()!.characters.split{$0==" "}
	for w in 0..<W {m[h][w]=Int(String(_a[w]))!}
}
var s=0
for i in 0..<Q {
	_a=readLine()!.characters.split{$0==" "}
	var w1=Int(String(_a[0]))!-1
	var h1=Int(String(_a[1]))!-1
	var w2=Int(String(_a[2]))!
	var h2=Int(String(_a[3]))!
	imos[h1][w1]+=1
	imos[h1][w2]-=1
	imos[h2][w1]-=1
	imos[h2][w2]+=1
}
for h in 0..<H {
	for w in 1..<W {imos[h][w]+=imos[h][w-1]}
}
for h in 1..<H {
	for w in 0..<W {imos[h][w]+=imos[h-1][w]}
}
for h in 0..<H {
	for w in 0..<W {
		if imos[h][w]>0 {s+=m[h][w]}
	}
}
print(s)
